src/com/projectkorra/projectkorra/storage/Database.java

```diff
@@ -116,6 +116,9 @@ public void run() {
 	 */
 	public ResultSet readQuery(String query) {
 		try {
+                        if(connection == null || connection.isClosed()) {
+                            open();

```

Comment:
You should probably reassign the connection variable to the new connection returned by open(). Otherwise it will still be using the old connection in the next line thus causing the same error